packages/compass-import-export/src/import/import-json.spec.ts

```diff
@@ -0,0 +1,544 @@
+import os from 'os';
+import assert from 'assert';
+import { BSONError, EJSON } from 'bson';
+import type { Document } from 'bson';
+import { MongoBulkWriteError } from 'mongodb';
+import fs from 'fs';
+import path from 'path';
+import { promisify } from 'util';
+import { Readable } from 'stream';
+import chai from 'chai';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import chaiAsPromised from 'chai-as-promised';
+import temp from 'temp';
+
+temp.track();
+
+import { DataServiceImpl } from 'mongodb-data-service';
+
+import { fixtures } from '../../test/fixtures';
+
+import type { ErrorJSON } from '../utils/import';
+
+import { guessFileType } from './guess-filetype';
+import { importJSON } from './import-json';
+
+const { expect } = chai;
+chai.use(sinonChai);
+chai.use(chaiAsPromised);
+
+describe('importJSON', function () {
+  let dataService: DataServiceImpl;
+  let dropCollection;
+  let createCollection;
+  let updateCollection: (ns: string, options: any) => Promise<Document>;
+
+  beforeEach(async function () {
+    dataService = new DataServiceImpl({
+      connectionString: 'mongodb://localhost:27018/local',
+    });
+
+    dropCollection = promisify(dataService.dropCollection.bind(dataService));
+
+    createCollection = promisify(
+      dataService.createCollection.bind(dataService)
+    );
+
+    updateCollection = promisify(
+      dataService.updateCollection.bind(dataService)
+    );
+
+    await dataService.connect();
+
+    try {
+      await dropCollection('db.col');
+    } catch (err) {
+      // ignore
+    }
+    await createCollection('db.col', {});
+  });
+
+  afterEach(async function () {
+    await dataService.disconnect();
+  });
+
+  for (const fixtureType of ['json', 'jsonl'] as const) {
+    for (const filepath of Object.values(fixtures[fixtureType])) {
+      const basename = path.basename(filepath);
+
+      it(`imports ${basename}`, async function () {
+        const typeResult = await guessFileType({
+          input: fs.createReadStream(filepath),
+        });
+        assert(typeResult.type === fixtureType);
+
+        const abortController = new AbortController();
+        const progressCallback = sinon.spy();
+
+        const ns = 'db.col';
+
+        const output = temp.createWriteStream();
+
+        const stats = await importJSON({
+          dataService,
+          ns,
+          input: fs.createReadStream(filepath),
+          output,
+          abortSignal: abortController.signal,
+          progressCallback,
+          jsonVariant: fixtureType,
+        });
+
+        expect(progressCallback.callCount).to.be.gt(0);
+
+        const totalRows = progressCallback.callCount;
+
+        expect(stats).to.deep.equal({
+          nInserted: totalRows,
+          nMatched: 0,
+          nModified: 0,
+          nRemoved: 0,
+          nUpserted: 0,
+          ok: Math.ceil(totalRows / 1000),
```

Comment:
Just out of curiosity, why this?

Reply:
The previously existing WritableCollectionStream has `ok` has the number of batches processed and the batch size is currently hardcoded as 1000.

Reply:
(What goes into that object is kinda the responsibility of WritableCollectionStream and its own tests, I just include this here as a sanity check and because I made its output also be the output of importJSON() so it is useful to have something fail if that changes because it would also change this interface.)

Reply:
I see. Thanks for explanation :)